### PR TITLE
Use reek directly

### PR DESCRIPTION
**Pull request by MGotink**
This contains the commits from and thus closes #255.

### CODE REVIEW

**Comment by MGotink** [comment]
@bf4 can you have a look at it before it's merged?

File: metric_fu.gemspec [hunk 1]
```diff
@@ -45,7 +45,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "flay",                  [">= 2.0.1",  "~> 2.1"]
   s.add_runtime_dependency "churn",                 ["~> 0.0.35"]
   s.add_runtime_dependency "flog",                  [">= 4.1.1",  "~> 4.1"]
-  s.add_runtime_dependency "reek",                  [">= 1.3.4",  "~> 1.3"]
+  s.add_runtime_dependency "reek",                  [">= 1.3.4",  "< 2.2"]
```
Comment (bf4) [inline metric_fu.gemspec]: why 2.2?

**Comment by bf4** [comment]
will do

File: metric_fu.gemspec (hunk 1 above)
Comment (MGotink) [inline metric_fu.gemspec]: Updated to 3.0, which makes more sense.

**Comment by bf4** [comment]
sorry, will do this weekend The wiki is maintained by the community. So if there aren't any up to date instructions, we recommend you to explore the solution yourself and hopefully contribute your findings back!

**Comment by MGotink** [comment]
@bf4, ok to merge?

**Comment by bf4** [comment]
looking at it now...  The wiki is maintained by the community. So if there aren't any up to date instructions, we recommend you to explore the solution yourself and hopefully contribute your findings back!

**Comment by bf4** [comment]
I created a PR in https://github.com/MGotink/metric_fu/pull/1 to fix reek compatibility but I'm not seeing the line numbers in the html reports... is that a regression from https://github.com/metricfu/metric_fu/pull/255 ?

**Comment by MGotink** [comment]
Thanks!
#255 adds line number to the yaml output, not the reports. Which still seems to work correctly.

**Comment by bf4** [comment]
I'm here if you want to chat I just invited you to https://gitter.im/metricfu/metric_fu?utm_source=share-link&utm_medium=link&utm_campaign=share-link

**Comment by bf4** [comment]
Ok, that's what I thought.  Should be good to merge, then. The wiki is maintained by the community. So if there aren't any up to date instructions, we recommend you to explore the solution yourself and hopefully contribute your findings back!

File: spec/metric_fu/metrics/reek/generator_spec.rb [hunk 2]
```diff
@@ -67,26 +47,56 @@
...
+                   smell_type: "Duplication",
+                   lines: [8]),
+
+            double(source: "app/controllers/newline_controller.rb",
+                   context: "NewlineController#some_method",
+                   message: "calls current_user.<< \"new line\n\" multiple times",
+                   smell_type: "Duplication",
+                   lines: [6, 9])
+        ]
+        @lines = double(smells: @smells)
```
Comment (bf4) [inline spec/metric_fu/metrics/reek/generator_spec.rb]: any reason we need to make the smells double and fill with with smell doubles?  I'd rather use verified doubles if we're going to do that... not a blocker, but a followup PR

Comment (MGotink) [inline spec/metric_fu/metrics/reek/generator_spec.rb]: That would be better yeah, I'll open a new PR for it.

**Comment by MGotink** [comment]
Merged, i think this can become 4.12?

**Comment by bf4** [comment]
Yup. You good with releasing it?

**Comment by bf4** [comment]
Oh, ref: using flay https://github.com/metricfu/metric_fu/pull/231

**Comment by MGotink** [comment]
Yeah I'm good with releasing it.
Was hoping to finish #260 quickly, but it's going to take some more time.

**Pull request merged**